client/components/language-picker/modal.jsx

```diff
@@ -417,7 +415,33 @@ export class LanguagePickerModal extends PureComponent {
 		this.handleClose();
 	};
 
-	handleClose = () => {
+	recordLanguagePickedEvent = ( isClosingWithoutSelection ) => {
+		const { selectedLanguageSlug, search } = this.state;
+
+		if (
+			isClosingWithoutSelection ||
+			this.initiallySelectedLanguageSlug === selectedLanguageSlug
+		) {
+			// we don't record an event if the language wasn't changed
+			return;
+		}
+
+		/**
+		 * We match specifically against the default value of `false`
+		 * rather than treating this falsy as `search` may be an empty
+		 * string if the user searches and then closes the search bar.
+		 * We'll naïvely assume that if someone searched *at all* that
+		 * we should record them as having used the search to find the
+		 * language they want. It's the simplest assumption for the
+		 * moment, and I'm not sure there's a better solution that
+		 * wouldn't be overkill given what we're trying to record here.
+		 */
+		const searched = false !== search;
+		this.props.recordTracksEvent( 'calypso_language_picker_language_picked', { searched } );
+	};
+
+	handleClose = ( isClosingWithoutSelection ) => {
```

Comment:
Should we add a default value here, because this is used without an argument in other locations? Given its usage right now, maybe it should be `false` by default?

Or alternatively, we could just call `handleClose` with an argument in all instances.

Reply:
Oh that’s a good point, I was just relying on it being `undefined` but that seems sloppy